File: ActivityTest/app/src/main/java/com/example/s405_00/activitytest/MainActivity.java
package com.example.s405_00.activitytest;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data)
    {
        if(resultCode == RESULT_OK)
        {
           String v1 = data.getStringExtra("c1");
           Toast.makeText(this,v1,Toast.LENGTH_SHORT).show();
           EditText editText = findViewById(R.id.editTextMain);
           editText.setText(data.getStringExtra("c2"));
        }

        super.onActivityResult(requestCode, resultCode, data);
    }

    public void onClick(View v)
    {
        EditText editText = findViewById(R.id.editTextMain);
        Intent i = new Intent(this,
                OtherActivity.class);
        i.putExtra("key1", "hello");
        i.putExtra("key2",editText.getText().toString() );
        //startActivity(i);
        startActivityForResult(i, 0);
    }
}

File: 이미지보내기 과제/study0116_Exam/app/src/main/java/com/example/dominico966/study0116_exam/MainActivity.kt
package com.example.dominico966.study0116_exam

import android.Manifest
import android.app.Activity
import android.content.Intent
import android.content.pm.PackageManager
import android.graphics.Bitmap
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v4.app.ActivityCompat
import android.support.v4.content.ContextCompat
import android.view.Menu
import android.view.MenuItem
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import okhttp3.*
import java.io.ByteArrayOutputStream
import java.io.File
import java.io.IOException
import javax.microedition.khronos.opengles.GL

class MainActivity : AppCompatActivity() {

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_layout, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when (item!!.itemId) {
            R.id.menu_item -> {
                val intent = Intent(this, SelectActivity::class.java)
                startActivity(intent)
            }
        }

        return super.onOptionsItemSelected(item)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.button_img_select).setOnClickListener {
            when (ContextCompat.checkSelfPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE)) {
                PackageManager.PERMISSION_DENIED -> {
                    ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE), 0)
                }

                PackageManager.PERMISSION_GRANTED -> {
                    val intent = Intent(Intent.ACTION_GET_CONTENT)
                    intent.type = "image/*"
                    startActivityForResult(intent, IMAGE_REQUEST)
                }
                else -> return@setOnClickListener
            }
        }
    }

    private val IMAGE_REQUEST = 1

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        when (requestCode) {
            0 -> {

            }

            IMAGE_REQUEST -> {
                when (resultCode) {
                    Activity.RESULT_OK -> {
                        val uri = data?.data
                        val realPath = RealPathUtil.getRealPath(this, uri)

                        (application as GlobalImages).realPath = realPath
                    }
                }
            }
            else -> return
        }
    }

}
